File: 42_GreatestSumOfSubarrays/GreatestSumOfSubarrays.cpp
#include <iostream>
#include <vector>
#include <string>
#include <deque>
#include <set>
#include <map>
#include <queue>
#include <stack>
#include <algorithm>
using namespace std;

class Solution
{
public:
    int FindGreatestSumOfSubArray(vector<int> array)
    {
        int curSum = array[0];
        int maxSum = array[0];
        for (int i = 1; i < array.size(); ++i)
        {
            curSum += array[i];
            if (maxSum < curSum)
                maxSum = curSum;
            if (curSum < 0)
                curSum = 0;
        }
        return maxSum;
    }
};

File: 42_GreatestSumOfSubarrays/GreatestSumOfSubarrays.py
class Solution:
    def FindGreatestSumOfSubArray(self, array):
        # write code here
        curSum, maxSum = 0, -float('inf')
        for i in array:
            curSum += i
            if curSum > maxSum:
                maxSum = curSum
            if curSum < 0:
                curSum = 0
        return maxSum
